node/node.go

```diff
@@ -93,3 +91,17 @@ func (n *Node) Stop(ctx context.Context) error {
 	log.Infof("%s Node is stopped", n.Type)
 	return nil
 }
+
+// newNode creates a new Node from given DI options.
+// DI options allow initializing the Node with a customized set of components and services.
+// NOTE: newNode is currently meant to be used privately to create various custom Node types e.g. full, unless we
+// decide to give package users the ability to create custom node types themselves.
+func newNode(opts ...fx.Option) (*Node, error) {
+	node := new(Node)
```

Comment:
It's not obvious in this PR' changeset to see where this `Node` actually gets initialized but that is probably due to diff only.

Reply:
Yeah, I just moved that below public methods